test/integration/router_test.go

```diff
@@ -521,20 +561,29 @@ func TestRouterPathSpecificity(t *testing.T) {
 				To: kapi.ObjectReference{
 					Name: "myService",
 				},
+				routeTLS: &routeapi.TLSConfig{
+					Termination:                   routeapi.TLSTerminationEdge,
+					Certificate:                   tr.ExampleCert,
+					Key:                           tr.ExampleKey,
+					CACertificate:                 tr.ExampleCACert,
+					InsecureEdgeTerminationPolicy: routeapi.InsecureEdgeTerminationPolicyAllow,
+				},
 			},
 		},
 	}
 	fakeMasterAndPod.RouteChannel <- eventString(routeEvent)
 
-	//ensure you can curl path and host
-	if err := waitForRoute(routeTestAddress, "www.example.com", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	if err := waitForRoute(routeAddress, "www.example.com", "http", nil, tr.HelloPod); err != nil {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	if err := waitForRoute(routeTestAddress, "www.example.com:80", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
+	for _, proto := range protocols {
+		//ensure you can curl path and host
+		if err := waitForRoute(routeTestAddress, "www.example.com", proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
+		if err := waitForRoute(routeAddress, "www.example.com", proto.name, nil, tr.HelloPod); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
+		if err := waitForRoute(routeTestAddress, "www.example.com:" + proto.port, proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
+		}
```

Comment:
same comments as earlier re: adding `protocol` information to the failure messages (the 3 `t.Fatalf`s in the for loop).